File: Curso Java Com POO/Java com POO/Aula10/src/aula10/Professor.java
package aula10;

public class Professor extends Pessoa {
    //Atributos
    private String especialidade;
    private float salario;
    
    //Métodos
    public void receberAum(float x) { //x = aumento
        this.setSalario(this.getSalario() + x);
        //this.salario += x;
    }
    
    //Getters e Setters
    //especialidade
    public String getEspecialidade() {
        return especialidade;
    }
    public void setEspecialidade(String especialidade) {
        this.especialidade = especialidade;
    }

    //salario
    public float getSalario() {
        return salario;
    }
    public void setSalario(float salario) {
        this.salario = salario;
    }
}

File: Curso Programação Orientada a Objetos/PHP com POO/Aula11/Professor.php
<?php

require_once 'Pessoa.php';

class Professor extends Pessoa {
    //Atributos
    private $especialidade;
    private $salario;
    
    //Métodos
    public function receberAumento($x) {
        echo "<p>Aumento no salário do professor " . $this->nome . "</p>";
        $this->setSalario($this->getSalario() + $x);
        //$this->salario += $x;
        echo "<p>Aumento de " . $x . "</p>";
        echo "<p>Salário agora é " . $this->getSalario() . "</p>";
    }
    
    //Getters
    public function getEspecialidade() {
        return $this->especialidade;
    }

    public function getSalario() {
        return $this->salario;
    }

    //Setters
    public function setEspecialidade($especialidade) {
        $this->especialidade = $especialidade;
    }

    public function setSalario($salario) {
        $this->salario = $salario;
    }


}
